ligo/stablecoin/actions.ligo

```diff
@@ -240,4 +333,236 @@ function is_operator_action
+      abs (current_allowance - unwrapped_parameter.amount)
+  ; const updated_allowances : minting_allowances =
+      Big_map.update (Tezos.sender, Some (updated_allowance), accumulator.minting_allowances)
+  ; const updated_store : storage = accumulator with record
+      [ minting_allowances = updated_allowances
+      ; total_supply = accumulator.total_supply + unwrapped_parameter.amount
+      ]
+  } with credit_to (unwrapped_parameter, updated_store)
+
+} with
+  ( (nil : list (operation))
+  , List.fold
+      ( function
+          ( const accumulator : storage
+          ; const mint_param  : mint_param
+          ) : storage is case accumulator.minting_allowances[senderAddress] of
+            None -> (failwith ("Sender address is not minter") : storage)
+          | Some (current_allowance) ->
+              mint_tokens (accumulator, mint_param, current_allowance)
+          end
+      , parameters
+      , store
+      )
+  )
+
+(*
+ * Decreases balance and total supply of tokens by the given amount
+ *)
+function burn
+  ( const parameters : burn_params
+  ; const store      : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+; const sender_address : address = Tezos.sender
+; function burn_tokens
+    ( const accumulator : storage
+    ; const burn_amount : nat
+    ) : storage is block
+  { const debited_storage : storage = debit_from
+      ( record
+          [ from_ = sender_address
+          ; amount = burn_amount
+          ]
+      , accumulator
+      )
+  } with debited_storage with record
+      [ total_supply = case is_nat (debited_storage.total_supply - burn_amount) of
+          Some (nat) -> nat
+        | None -> (failwith ("negative total supply") : nat)
+        end
+      ]
+} with
+  ( (nil : list (operation))
+  , List.fold (burn_tokens, parameters, store)
+  )
+
+(*
+ * Sets contract owner to a new address
+ *)
+function transfer_ownership
+  ( const parameter : transfer_ownership_param
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_owner (store)
+} with
+  ( (nil : list (operation))
+  , store with record [roles.pending_owner = Some (parameter)]
+  )
+
+(*
+ * When pending sender calls this entrypoint, the address
+ * receives owner privileges for contract
+ *)
+function accept_ownership
+  ( const parameter : accept_ownership_param
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_pending_owner (store)
+} with
+  ( (nil : list (operation))
+  , store with record
+      [ roles.pending_owner = (None : option (address))
+      // We have already validated pending owner
+      ; roles.owner = Tezos.sender
+      ]
+  )
+
+(*
+ * Moves master minter priveleges to a new address.
+ * Fails if sender is not contract owner.
+ *)
+function change_master_minter
+  ( const parameter : change_master_minter_param
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_master_minter (store)
```

Comment:
It looks wrong. It will check that the sender is the current master minter, but it should check that the sender is the current contract owner. Same for `change_pauser`.

Reply:
Apparently tests are also broken.

Reply:
Curious. It seems that it was `authorize_owner` for both `change_pauser` and `change_master_minter` entrypoints in the commit that I reviewed.

https://github.com/tqtezos/stablecoin/blob/2866d4fa4884ebd1c0de85c856dda5109a33c809/ligo/stablecoin/actions.ligo#L550